internal/controller/drplacementcontrol.go

```diff
@@ -1186,22 +1194,48 @@ func (d *DRPCInstance) readyToSwitchOver(homeCluster string, preferredCluster st
 		}
 	}
 	// Allow switch over when PV data is ready and the cluster data is protected
-	return d.isVRGConditionMet(homeCluster, VRGConditionTypeDataReady) &&
-		d.isVRGConditionMet(homeCluster, VRGConditionTypeClusterDataProtected)
+	dataReady, err := d.isVRGConditionMet(homeCluster, VRGConditionTypeDataReady)
+	if err != nil || !dataReady {
+		return false
+	}
```

Comment:
We need to log the error or pass it to caller. Dropping an error silently without any check is not an option. How are we going to debug the error?

Since we we don't use dataReady after this check, we should keep dataReady and err in the scope of the if block:

```go
if dataReady, err := d.isVRGConditionMet(homeCluster, VRGConditionTypeDataReady); err != nil || !dataReady {
    return false
}
```

This avoids the confusion about err value when we return below.

Reply:
Done